.github/workflows/deploy.yml

```diff
@@ -1,25 +1,54 @@
-name: Deploy
+name: Compile and Deploy
 
 on:
   push:
     branches:
       - main
 
 jobs:
-  build:
 
+  compile:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Check out code
+      - uses: actions/checkout@v2
+
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Process translations
+        run: |
+          ./bw-dev messages:update
+
+      - name: Compile site
+        run: |
+          python generate.py
+
+  deploy:
+    needs: compile
```

Comment:
What I'm trying to do here, and I _think_ it's right, is to say "if `compile` fails, don't run `deploy` action": https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#defining-prerequisite-jobs

Reply:
Eh, yes, absolutely not only about ordering. But what will be rsync'd?

Reply:
I didn't touch that but everything in `/site` will be rsync'd to the server. That is currently happening, the problem is that we're not compiling anything, so rsync doesn't do anything because nothing changed in the `/site` directory.